src/app/modules/registration/pages/obs-location/obs-location.page.ts

```diff
@@ -44,35 +46,43 @@ export class ObsLocationPage implements OnInit, OnDestroy {
 
   async ngOnInit() {
     const id = this.activatedRoute.snapshot.params['id'];
+    const geoHazard = this.activatedRoute.snapshot.params['geoHazard'];
+    const lat = this.activatedRoute.snapshot.queryParams['lat'];
+    const lon = this.activatedRoute.snapshot.queryParams['lon'];
+    const locationId = this.activatedRoute.snapshot.queryParams['locationId'];
     if (id) {
       // Edit an existing draft
       this.draft = await this.draftService.load(id);
       this.geoHazard = this.draft.registration.GeoHazardTID;
-    } else if (this.activatedRoute.snapshot.params['geoHazard']) {
+    } else if (geoHazard) {
       // New draft - will be created later
-      this.geoHazard = parseInt(this.activatedRoute.snapshot.params['geoHazard'], 10);
+      if (isNaN(geoHazard)){
+        const geoHazardCapLetter = geoHazard.charAt(0).toUpperCase() + geoHazard.slice(1);
+        this.geoHazard = parseInt(GeoHazard[geoHazardCapLetter], 10);
+      } else {
+        this.geoHazard = parseInt(geoHazard, 10);
+      }
     }
     if (this.geoHazard == null) {
       // No geohazard found, use app mode
       const userSettings = await firstValueFrom(this.userSettingService.userSetting$);
       this.geoHazard = userSettings.currentGeoHazard[0];
     }
+    if(lat && lon){
+      this.setLocationMarker(lat, lon);
```

Comment:
Det ser ut som vi setter location marker ganske mange ganger her? Hvis `lat`, `lon`, `locationId` og `this.hasLocation(this.draft)` er `true` så kaller vi `this.setLocationMarker(lat, lon)` 3 ganger.